src/propTypes/albumTypes.js

```diff
@@ -0,0 +1,28 @@
+import PropTypes from 'prop-types';
+
+export const albumShape = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  albumName: PropTypes.string.isRequired,
+  artistName: PropTypes.string.isRequired,
+  image: PropTypes.string.isRequired,
+  // custom validator function takes three arguments
+  releaseDate: (props, propName, componentName) => {
+    if (!/^\d{4}-\d{2}-\d{2}$/.test(props[propName])) {
+      // format YYYY-MM-DD
+      return new Error(
+        `Invalid prop ${propName} supplied to ${componentName}. Validation failed.`
+      );
+    }
+  },
```

Comment:
Out backend [model](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/1e48ca722a1a852752cda040cae126beeffd535d/src/models/Album.js#L30) says this field is not required. So it's possible sometimes it will be null/undefined. Currently, this code means it would fail validation in that case (because the regex wouldn't match). But we probably just want to skip checking the regex in that case and treat the value as allowed/valid.

```suggestion
releaseDate: (props, propName, componentName) => {
    if (props[propName]) {
      if (!/^\d{4}-\d{2}-\d{2}$/.test(props[propName])) {
        // format YYYY-MM-DD
        return new Error(
          `Invalid prop ${propName} supplied to ${componentName}. Validation failed.`
        );
      }
    }
  },
```